shell/platform/darwin/macos/BUILD.gn

```diff
@@ -334,3 +337,21 @@ generated_file("macos_framework_without_entitlement_config") {
 
   deps = [ ":_generate_symlinks" ]
 }
+
+shared_library("_generate_symlinks_and_verify_framework_module") {
+  framework_search_path = rebase_path("$root_out_dir")
+  visibility = [ ":*" ]
+  cflags_objc = [ "-F$framework_search_path" ]
+
+  ldflags = [
+    "-F$framework_search_path",
+    "-fapplication-extension",
+    "-Xlinker",
+    "-fatal_warnings",
+  ]
+  frameworks = [ "FlutterMacOS.framework" ]
+
+  sources = [ "framework/Source/FlutterUmbrellaImport.m" ]
+
+  deps = [ ":_generate_symlinks" ]
```

Comment:
This is the test. @Hixie 

This build target will fail if the FlutterMacOS framework is not application extension safe.

Reply:
If you haven't yet you'll need to ping him in Discord in the #hackers channel.

You could also rename t`FlutterUmbrellaImport` to something that would indicate it's a test https://github.com/flutter/cocoon/blob/412b236f4b16175e9459cdcff430147da4f6ff37/app_dart/lib/src/request_handlers/github/webhook_subscription.dart#L35